clang/lib/Parse/Parser.cpp

```diff
@@ -1473,7 +1475,7 @@ Decl *Parser::ParseFunctionDefinition(ParsingDeclarator &D,
   D.getMutableDeclSpec().abort();
 
   if (BodyKind != Sema::FnBodyKind::Other) {
-    Actions.SetFunctionBodyKind(Res, KWLoc, BodyKind);
+    Actions.SetFunctionBodyKind(Res, KWLoc, BodyKind, DeletedMessage);
```

Comment:
Here we might want just want to do Res->setDeletedMessage(), or something along these lines?

Reply:
Hmm, true; it might also be useful to just have that already available anyway in case we end up needing it elsewhere.

Reply:
So after looking into this a bit, I don’t think we want to do that because `Res` is not a `FunctionDecl*` here, it’s a `Decl*`. Specifically, it could also be a template, which means we’d need to first dig out the actual function using `Sema::AdjustDeclIfTemplate`; the same is also true in at least one other place. `SetFunctionBodyKind` calls `SetDeclDeleted`, which already takes care of all of that, so I think it’d make more sense to put that there (however, a `setDeletedMessage()` on `FunctionDecl` is still useful, and I’ll still be using that in other places).